classes/04class/exercises/c04-iac04/raghunadhpokkalath/terraform-code/iac-04-module/main.tf

```diff
@@ -0,0 +1,126 @@
+resource "aws_security_group" "allow_http" {
+  name   = "allow_http"
+  vpc_id = var.vpc.id
+  ingress {
+    description = "Allow http connection"
+    from_port   = 80
+    to_port     = 80
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = {
+    Name = var.tags.alb_sg
+  }
+}
+
+resource "aws_security_group" "allow_only_alb" {
+    vpc_id = var.vpc.id   
+  name   = "allow_only_alb"
+  ingress {
+    description = "Allow only ALB connection to EC2"
+    from_port   = 80
+    to_port     = 80
+    protocol    = "tcp"  
+    security_groups = [aws_security_group.allow_http.id]
+  }
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+  tags = {
+    Name = var.tags.ec2_sg
+  }
+}
+
+
+
+
+resource "aws_lb" "this" {
+  name               = "devops-academy-iac-lb"
+  internal           = "false"
+  load_balancer_type = "application"
+  security_groups    = [aws_security_group.allow_http.id]
+  subnets            = var.public_subnet_ids
+
+  tags = {
+    Name = var.tags.alb
+  }
+}
+
+resource "aws_lb_target_group" "this" {
+  name     = "devops-academy-iac-lb-tg"
+  port     = 80
+  protocol = "HTTP"
+  vpc_id   = var.vpc.id
+}
+
+resource "aws_lb_listener" "this" {
+  load_balancer_arn = aws_lb.this.arn
+  port              = "80"
+  protocol          = "HTTP"
+
+  default_action {
+    type             = "forward"
+    target_group_arn = aws_lb_target_group.this.arn
+  }
+}
+
+
+
+resource "aws_launch_configuration" "this" {
+  name_prefix = "devopsacademy-iac-launchconfig"
+  image_id      = var.ami_id
+  instance_type = "t2.micro"
+  key_name = "EC2KeyPair"
+  user_data_base64 =  base64encode(var.user_data)
```

Comment:
can you change user_data to come from a file or write it inline here?

Reply:
Thanks @denstorti moved the user data to user_data.sh file